src/app/dashboard/_components/Category.tsx

```diff
@@ -21,7 +22,15 @@ function Category({ category }: CategoryProps) {
     setIsEditingTitle(true);
   };
 
-  const handleEditFinish = () => {
+  const handleEditFinish = async () => {
+    if (!dbInstance) return;
+
+    await categoryRepository.editCategory(
+      dbInstance,
+      category.id,
+      newCategoryTitle,
+    );
+    fetchCategories();
```

Comment:
dbInstance를 Category가 몰라도 되도록 onEditFinish라는 props를 부모에서 넘겨주고 이 함수에서 repository.editCategory랑 fetchCategories를 실행하면 어떤가요? 

```suggestion
   await onEditFinish(category.id, newCategoryTitle)
```